### PR TITLE
samples: clean up tests

### LINKED ISSUE #202: samples.snippets.iam_test: test_check_subscription_permissions failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: 9667ef10dd9364b6a50ede02907dd5cc64e9109c
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/1300bb80-1aa1-4ffd-b951-3cc6ab23fabd), [Sponge](http://sponge2/1300bb80-1aa1-4ffd-b951-3cc6ab23fabd)
status: failed
<details><summary>Test output</summary><br><pre>args = (resource: "projects/python-docs-samples-tests/subscriptions/iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5"
permissions: "pubsub.subscriptions.consume"
permissions: "pubsub.subscriptions.update"
,)
kwargs = {'metadata': [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/subscriptions/iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5'), ('x-goog-api-client', 'gl-python/3.6.10 grpc/1.32.0 gax/1.22.2')]}

    @six.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
>           return callable_(*args, **kwargs)

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <grpc._channel._UnaryUnaryMultiCallable object at 0x7f6888d40828>
request = resource: "projects/python-docs-samples-tests/subscriptions/iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5"
permissions: "pubsub.subscriptions.consume"
permissions: "pubsub.subscriptions.update"

timeout = None
metadata = [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/subscriptions/iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5'), ('x-goog-api-client', 'gl-python/3.6.10 grpc/1.32.0 gax/1.22.2')]
credentials = None, wait_for_ready = None, compression = None

    def __call__(self,
                 request,
                 timeout=None,
                 metadata=None,
                 credentials=None,
                 wait_for_ready=None,
                 compression=None):
        state, call, = self._blocking(request, timeout, metadata, credentials,
                                      wait_for_ready, compression)
>       return _end_unary_response_blocking(state, call, False, None)

.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py:826: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

state = <grpc._channel._RPCState object at 0x7f6888d408d0>
call = <grpc._cython.cygrpc.SegregatedCall object at 0x7f6888dab988>
with_call = False, deadline = None

    def _end_unary_response_blocking(state, call, with_call, deadline):
        if state.code is grpc.StatusCode.OK:
            if with_call:
                rendezvous = _MultiThreadedRendezvous(state, call, None, deadline)
                return state.response, rendezvous
            else:
                return state.response
        else:
>           raise _InactiveRpcError(state)
E           grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
E           	status = StatusCode.NOT_FOUND
E           	details = "Resource not found (resource=iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5)."
E           	debug_error_string = "{"created":"@1600506354.060659276","description":"Error received from peer ipv4:74.125.142.95:443","file":"src/core/lib/surface/call.cc","file_line":1061,"grpc_message":"Resource not found (resource=iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5).","grpc_status":5}"
E           >

.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py:729: _InactiveRpcError

The above exception was the direct cause of the following exception:

subscription = 'projects/python-docs-samples-tests/subscriptions/iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5'
capsys = <_pytest.capture.CaptureFixture object at 0x7f6888d33780>

    def test_check_subscription_permissions(subscription, capsys):
>       iam.check_subscription_permissions(PROJECT, SUBSCRIPTION)

iam_test.py:117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
iam.py:171: in check_subscription_permissions
    request={"resource": subscription_path, "permissions": permissions_to_check}
../../google/cloud/pubsub_v1/_gapic.py:40: in <lambda>
    fx = lambda self, *a, **kw: wrapped_fx(self.api, *a, **kw)  # noqa
../../google/pubsub_v1/services/subscriber/client.py:1896: in test_iam_permissions
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,)
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/gapic_v1/method.py:145: in __call__
    return wrapped_func(*args, **kwargs)
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:59: in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = None
from_value = <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "Resource not found (resourc...pc_message":"Resource not found (resource=iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5).","grpc_status":5}"
>

>   ???
E   google.api_core.exceptions.NotFound: 404 Resource not found (resource=iam-test-subscription-4b05dc75e0544930b8c12fe1323593d5).

<string>:3: NotFound</pre></details>

### LINKED ISSUE #217: samples.snippets.publisher_test: test_resume_publish_with_error_handler failed
This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: 7228f6c9a4c050bf22bb4bc3582b89b04eaa8702
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/92c63409-6174-42e0-b2d4-a04524103ae6), [Sponge](http://sponge2/92c63409-6174-42e0-b2d4-a04524103ae6)
status: failed
<details><summary>Test output</summary><br><pre>args = (topic: "projects/python-docs-samples-tests/topics/publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c"
,)
kwargs = {'metadata': [('x-goog-request-params', 'topic=projects/python-docs-samples-tests/topics/publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c'), ('x-goog-api-client', 'gl-python/3.6.10 grpc/1.32.0 gax/1.22.2')]}

    @six.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
>           return callable_(*args, **kwargs)

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <grpc._channel._UnaryUnaryMultiCallable object at 0x7f115f744da0>
request = topic: "projects/python-docs-samples-tests/topics/publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c"

timeout = None
metadata = [('x-goog-request-params', 'topic=projects/python-docs-samples-tests/topics/publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c'), ('x-goog-api-client', 'gl-python/3.6.10 grpc/1.32.0 gax/1.22.2')]
credentials = None, wait_for_ready = None, compression = None

    def __call__(self,
                 request,
                 timeout=None,
                 metadata=None,
                 credentials=None,
                 wait_for_ready=None,
                 compression=None):
        state, call, = self._blocking(request, timeout, metadata, credentials,
                                      wait_for_ready, compression)
>       return _end_unary_response_blocking(state, call, False, None)

.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py:826: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

state = <grpc._channel._RPCState object at 0x7f115c746b00>
call = <grpc._cython.cygrpc.SegregatedCall object at 0x7f115c738e08>
with_call = False, deadline = None

    def _end_unary_response_blocking(state, call, with_call, deadline):
        if state.code is grpc.StatusCode.OK:
            if with_call:
                rendezvous = _MultiThreadedRendezvous(state, call, None, deadline)
                return state.response, rendezvous
            else:
                return state.response
        else:
>           raise _InactiveRpcError(state)
E           grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
E           	status = StatusCode.NOT_FOUND
E           	details = "Resource not found (resource=publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c)."
E           	debug_error_string = "{"created":"@1601629662.820561737","description":"Error received from peer ipv4:173.194.202.95:443","file":"src/core/lib/surface/call.cc","file_line":1061,"grpc_message":"Resource not found (resource=publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c).","grpc_status":5}"
E           >

.nox/py-3-6/lib/python3.6/site-packages/grpc/_channel.py:729: _InactiveRpcError

The above exception was the direct cause of the following exception:

client = <google.cloud.pubsub_v1.PublisherClient object at 0x7f115f761588>

    @pytest.fixture
    def topic_publish(client):
        topic_path = client.topic_path(PROJECT, TOPIC_PUBLISH)
    
        try:
            topic = client.get_topic(request={"topic": topic_path})
        except:  # noqa
            topic = client.create_topic(request={"name": topic_path})
    
        yield topic.name
    
>       client.delete_topic(request={"topic": topic.name})

publisher_test.py:61: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../google/cloud/pubsub_v1/_gapic.py:40: in <lambda>
    fx = lambda self, *a, **kw: wrapped_fx(self.api, *a, **kw)  # noqa
../../google/pubsub_v1/services/publisher/client.py:883: in delete_topic
    request, retry=retry, timeout=timeout, metadata=metadata,
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/gapic_v1/method.py:145: in __call__
    return wrapped_func(*args, **kwargs)
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/retry.py:286: in retry_wrapped_func
    on_error=on_error,
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/retry.py:184: in retry_target
    return target()
.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:59: in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = None
from_value = <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "Resource not found (resourc...age":"Resource not found (resource=publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c).","grpc_status":5}"
>

>   ???
E   google.api_core.exceptions.NotFound: 404 Resource not found (resource=publisher-test-topic-publish-19fde27fa8bd4912842ab66acabb8a9c).

<string>:3: NotFound</pre></details>

### PR DESCRIPTION
**Pull request by anguillanneuf**
Fixes #202 
Fixes #217 

After adding samples for publishing messages with ordering keys, they take longer to run. Increase max timeout for tests to fix flaky tests (likely due to multiple test sessions in parallel).

I took the opportunity to clean up tests and samples as well. 
- Moved subscription detachment sample to from iam.py to publisher.py because we use the publisher client to detach subscriptions.
- Rewrote synchronous pull with lease management sample to not use a worker function as it would cause errors in python3.8. 
- Used f"" in print strings instead of "".format(). 
- Standardized resource creation and deletion tests so they don't step on each other.  
- Ran `black . -l 88` for lint.

### CODE REVIEW

**Review by tmatsuo — COMMENTED** [review]
`subscription` fixture doesn't have scope specified. Do you know the default value?
How about to add an explicit scope for avoiding errors due to scope?

File: samples/snippets/iam_test.py [hunk 1]
```diff
@@ -24,6 +25,8 @@
 PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
 TOPIC_ID = "iam-test-topic-" + UUID
 SUBSCRIPTION_ID = "iam-test-subscription-" + UUID
+# Allow 60 s for tests to finish.
+max_time = 60
```
Comment (tmatsuo) [inline samples/snippets/iam_test.py:29]: For consistency, maybe `MAX_TIME` instead of `max_time`?

File: samples/snippets/publisher_test.py [hunk 2]
```diff
@@ -17,48 +17,65 @@
...
 
 UUID = uuid.uuid4().hex
-PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
-TOPIC_ADMIN = "publisher-test-topic-admin-" + UUID
-TOPIC_PUBLISH = "publisher-test-topic-publish-" + UUID
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+TOPIC_ID = "publisher-test-topic-" + UUID
+SUBSCRIPTION_ID = "publisher-test-subscription-" + UUID
+# Allow 60s for tests to finish.
+max_time = 60
```
Comment (tmatsuo) [inline samples/snippets/publisher_test.py:32]: Nit: maybe `MAX_TIME` instead of `max_time`?

File: samples/snippets/subscriber.py [hunk 3]
```diff
@@ -524,89 +518,63 @@ def synchronous_pull_with_lease_management(project_id, subscription_id):
     # [START pubsub_subscriber_sync_pull_with_lease]
     import logging
     import multiprocessing
-    import random
+    import sys
     import time
 
     from google.cloud import pubsub_v1
 
+    multiprocessing.log_to_stderr()
```
Comment (tmatsuo) [inline samples/snippets/subscriber.py:526]: Do you want to show these logging setup in our documentation?

Comment (anguillanneuf) [reply samples/snippets/subscriber.py:526]: Yes. It makes sense for this sample. The idea is to pull messages synchronously, take one and process them in a different process (or thread), extend each message's acknowledgement deadline as it is being processed. Logging here helps user keep track of which process is handling which message.

Comment (tmatsuo) [reply samples/snippets/subscriber.py:526]: Thanks

File: samples/snippets/iam_test.py [hunk 4]
```diff
@@ -54,77 +56,57 @@ def subscriber_client():
...
+        request={"name": subscription_path, "topic": topic_path}
+    )
+    yield subscription.name
 
     try:
         subscriber_client.delete_subscription(
             request={"subscription": subscription_path}
         )
-    except Exception:
+    except NotFound:
```
Comment (tmatsuo) [inline samples/snippets/iam_test.py:71]: Thanks, it's much better.

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**